slash-commands/slash-commands.js

```diff
@@ -115,14 +149,23 @@ CandyShop.SlashCommands = (function(self, Candy, $) {
 	self.part = function() {
 		Candy.Core.Action.Jabber.Room.Leave(self.currentRoom());
 	};
+	
+	/** Function: leave
+	 * Exits the current chat room
+	 *
+	 */
+	self.leave = function() {
+		Candy.Core.Action.Jabber.Room.Leave(self.currentRoom());
+	};
 
 	/** Function: topic
 	 * Sets the topic (subject) for the current chat room
 	 *
 	 * Parameters:
-	 * 	(String) topic The new topic for the room
+	 *    (String) topic The new topic for the room
 	 */
 	self.topic = function(topic) {
+		// TODO: .isModerator() && allowed checks
```

Comment:
No prior check doesn’t imply silent failure, as the XEP mandates a <forbidden/> to be returned (with some explanation text maybe), which you should then display to the user as the result of the failed topic change.